synapse/handlers/sync.py

```diff
@@ -630,10 +633,17 @@ async def _load_filtered_recents(
 
             prev_batch_token = now_token.copy_and_replace("room_key", room_key)
 
+        # Don't bother to bundle aggregations if the timeline is unlimited,
+        # as clients will have all the necessary information.
```

Comment:
Is the intent here to let e.g. let clients show accurate reaction counts, even if some of the reaction events were skipped over in a gappy sync?

Reply:
Yes, pretty much.

Reply:
(I should note that this comment was copied from `synapse/rest/client/sync.py`, but if we can make it clearer, please let me know!)